kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -429,7 +428,13 @@ def _retrieve_stats(self):
             return {}
 
     def _report_node_metrics(self, instance_tags):
-        node_spec = self._retrieve_node_spec()
+        try:
+            node_spec = self._retrieve_node_spec()
+        except requests.HTTPError:
+            # ignore HTTPError, for supporting k8s 1.18 in a degrated mode
+            self.log.warning('kubelet check "/spec" endpoint not available')
```

Comment:
This would suck for users on 1.18+. Can we detect the version?

Reply:
Unfortunately we can't have the version from the kubelet. Also the `/spec` endpoint can be reactivated by a kubelet configuration change. In kubernetes 1.19 this endpoint will be completely removed.